### DISCUSSION TITLE
Does Qdrant support a 'partitioning' feature similar to Milvus that can enhance query performance?

### DISCUSSION THREAD

**Discussion by Emma-Miao** [qdrant/qdrant#4401] [Q&A] [answered]
Does Qdrant have the ability to divide the physical storage of a collection into multiple parts based on one specific field, like 'partition key' in milvus?

**Comment by timvisee** [comment]
At this point, no.

However, you _could_ separate different collections across different drives by using different mounts.

**Reply by Emma-Miao** [reply]
Thanks for your reply. Do you have any plan to add such functionality? I want to store about 2500million user data, and each of them have thousands vectors, if all of a user's vector data can be stored in contiguous physical space, i think it will lead to a significant improvement in filtering performance.

**Reply by generall** [reply, marked as answer]
I don't think `partition key` feature actually stores data on different physical storages.

If you need to store 2500million vectors in qdrant, it is totally possible with sharding and multitenancy.

https://qdrant.tech/documentation/guides/multiple-partitions/

https://qdrant.tech/documentation/guides/distributed_deployment/#sharding
